local-cluster/src/cluster_tests.rs

```diff
@@ -284,6 +284,26 @@ pub fn kill_entry_and_spend_and_verify_rest(
     }
 }
 
+pub fn check_for_new_roots(num_new_roots: usize, contact_infos: &[ContactInfo]) {
+    let mut roots = vec![HashSet::new(); contact_infos.len()];
+    let mut done = false;
+    let mut last_print = Instant::now();
+    while !done {
+        for (i, ingress_node) in contact_infos.iter().enumerate() {
+            let client = create_client(ingress_node.client_facing_addr(), VALIDATOR_PORT_RANGE);
+            let slot = client.get_slot().unwrap_or(0);
+            roots[i].insert(slot);
+            let min_node = roots.iter().map(|r| r.len()).min().unwrap_or(0);
+            if last_print.elapsed().as_secs() > 3 {
+                info!("PARTITION_TEST min observed roots {}/16", min_node);
+                last_print = Instant::now();
+            }
+            done = min_node >= num_new_roots;
+        }
+    }
+    sleep(Duration::from_millis(clock::DEFAULT_MS_PER_SLOT / 2));
```

Comment:
shouldn't `sleep` be inside `while !done { ... }`?

Reply:
yup oops! bad copy and paste from here: https://github.com/solana-labs/solana/pull/9805/files#diff-a6e7be832768ca536ac5b7450a42ee09L366